instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -41,6 +41,23 @@ def request(req, body = nil, &block)
 
             private
 
+            def instrumentation_config
+              Net::HTTP::Instrumentation.instance.config
+            end
+
+            def inject_context_and_extract_data(span, req, body)
+              OpenTelemetry.propagation.inject(req)
+              safe_execute_hook(instrumentation_config[:request_hook], span, req, body)
+            end
+
+            def safe_execute_hook(hook, *args)
+              return if hook.nil?
+
+              hook.call(*args)
+            rescue StandardError => e
+              OpenTelemetry.logger.debug(e.message)
```

Comment:
Did you consider using [`OpenTelemetry.handle_error`](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/api/lib/opentelemetry.rb#L44)?

The advantage of using the `handle_error` is that it provides access to the `Error` and make it possible to inspect details like the backtrace.

Reply:
Done